src/containers/OrgDetails.tsx

```diff
@@ -107,71 +142,53 @@ export const OrgDetails = () => {
         return true;
       })
   });
-
-  const methods = useForm<OrganizerInfo>({
+  const methods = useForm<OrganizerInformationForm>({
     resolver: yupResolver(organizerSchema),
-    defaultValues: groupCheckout?.organizerInfo ?? defaultValues
+    defaultValues: defaultValuesSessionStorage || defaultValues
   });
+  const { watch, handleSubmit } = methods;
+  const { privacy } = watch();
 
-  const {
-    watch,
-    handleSubmit,
-    formState: { isSubmitting }
-  } = methods;
-
+  /**
+   * In this useEffect data is being stored in the session storage state,
+   * while a user is updating the form.
+   */
   useEffect(() => {
     const subscription = watch((values) => {
-      dispatch({
-        type: 'SET_GROUP_CHECKOUT',
-        payload: {
-          ...groupCheckout,
-          organizerInfo: {
-            ...groupCheckout?.organizerInfo,
-            ...values
-          }
-        } as GroupCheckOut
-      });
+      const { privacy: _, ...rest } = values;
+      const formattedOrganizerInfo = getFormattedOrganizerInfo(
+        rest as OrganizerInformationForm
```

Comment:
`as` should not be used it is a workaround, functions that returns values should have output typed